### PR TITLE
PP-8191 add switch payment provider service

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- add a new service that handles switch payment provider service request
- add tests

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gatewayaccount/exception/SwitchPaymentProviderException.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.pay.connector.gatewayaccount.exception;
+
+import javax.ws.rs.BadRequestException;
+
+public class SwitchPaymentProviderException extends BadRequestException {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/exception/SwitchPaymentProviderException.java]: As this is a fairly generic exception, do you think there's much point in having a dedicated exception class? Would it make more sense to just throw a `new WebApplicationException(badRequestResponse("message"))` as we do in various other places?
Or just a `BadRequestException`?

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        this.gatewayAccountCredentialsDao = gatewayAccountCredentialsDao;
+    }
+
+    @Transactional
+    public void switchPaymentProviderForAccount(GatewayAccountEntity gatewayAccountEntity,
+                                                GatewayAccountSwitchPaymentProviderRequest request) {
+
+        List<GatewayAccountCredentialsEntity> credentials = gatewayAccountEntity.getGatewayAccountCredentials();
+
+        if (credentials.size() != 2) {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java]: I don't think we should rely on only having exactly 2 credentials to future proof, I think we should only throw if there are fewer than 2.

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            throw new SwitchPaymentProviderException("Account has no credential to switch to/from");
+        }
+
+        var activeCredentialEntity = credentials
+                .stream()
+                .filter(credential -> ACTIVE.equals(credential.getState())).findFirst()
+                .orElseThrow(() -> new NotFoundException(format("Account credential with id ACTIVE state not found.", request.getGatewayAccountCredentialId())));
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java]: I'm not sure a 404 makes sense here, as it's usually used to suggest that either the endpoint wasn't found or a resource specified in the request wasn't found. I think 400 makes more sense.

**Review by stephencdaly — COMMENTED** [review]
Looks good, just a few small comments about exceptions.

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java (hunk 3 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java]: Story -> scenario 6

Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java]: The story says 400

Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java]: Also I think the error message is a bit wrong, I think " id" should be removed

Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountSwitchPaymentProviderService.java]: 🤦

**Review by stephencdaly — APPROVED** [review]
LGTM!

**Pull request merged**